variant-inc/terragrunt-variant-apps
# Feature/cloud 1751 pin version numbers

# Description

Pin version numbers to terraform modules

## Type of change

Please delete options that are not relevant.

(fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

```yaml
Pre reqs:
  - Create a branch in demo flask api and add postgress database https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml#L10
  - Create a commit and deploy (sample: https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/demo-test-postgress/deployments/releases/0.1.0-test-master-post0001.89/deployments/Deployments-38114?activeTab=taskLog)

Test Case 1 [Verify app is deployed and nothing is breaking]:
- Result
 - App should be deployed in cluster

Test Case 2 [See database is created and run read queries with the role]:
- Steps
  - run /database/read
- Results
  - Should return true as response
  - Check pod logs for results from select comand

Test Case 2 [See database is created and run write queries with the role]:
- Steps
  - run /database/write
- Results
  - Should return true as response
  - Check pod logs for results from table creation and insert results
  
Test Case 3 [Delete the records insert]:
- Steps
  - run /database/delete
- Results
  - Should return true as response
  - Check pod logs for results from table and records deletion

Test Case 4 [To see tags version pinning is working]:
- Steps
  - Make sure app is deployed
- Results
  - check in app role tags to see tags are added as given in api.yaml

Test Case 5 [To create s3 bucket]:
- Steps
  - Make sure app is deployed
- Results
  - check bucket is created

Test Case 6 [To create SNS Topic]:
- Steps
  - Make sure app is deployed
  - With the given sns topic spec https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml
- Results
  - Should see sns got created.

Test Case 7 [To create SNS/SQS subscription]:
- Steps
  - Make sure app is deployed
  - With the given sns topic spec https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml
- Results
  - Should see sqs got created.
  - Should see sns/sqs subscription got created.
 
Test Case 7 [Run SNS/SQS endpoints]:
- Steps
  - Make sure app is deployed
  - With the given sns topic spec https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml
  - run /sns endpoint
  - run /sqs endpoint(Should add some message to external topic so that sqs can pull that message after subscription)

- Results
  - Should see message got posted in the pod logs.
  - Expect some some delay to receive response after post message to externat topic , when /sqs endpoint has ran

 
Test Case 8 [Check Dlq created]:
- Steps
  - Make sure app is deployed
  - With the given sns topic spec https://github.com/variant-inc/demo-python-flask-variant-api/blob/test/master-postgress/.variant/deploy/base-api.yaml
- Results
  - Check dlq is created

Test Case 9 [Deploy Dynamo db/ run endpoint]:
- Steps
  - Make sure app is deployed
  - With the given dynamo db spec https://github.com/variant-inc/demo-python-flask-variant-api/tree/test/dynamodb
  - octopus link https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/demo-test-dynamodb/deployments
  - run /dynamo-insert endpoint
- Results
  - Should see table got created and record got inserted.
```


## Checklist
